http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -173,8 +156,63 @@ void sendMessage(final String message) {
       if (!connectionIsOpen) {
         queuedMessages.add(message);
       } else {
-        client.send(message);
+        client
+            .sendText(message, true)
+            .exceptionally(logWebSocketError(Level.SEVERE, "Failed to send text."));
       }
     }
   }
+
+  private <T> Function<Throwable, T> logWebSocketError(
+      final Level level, final String errorMessage) {
+    return throwable -> {
+      log.log(level, errorMessage, throwable);
+      return null;
+    };
+  }
+
+  @VisibleForTesting
+  class WebSocketListener implements Listener {
+    private final StringBuilder textAccumulator = new StringBuilder();
+
+    @Override
+    public void onOpen(final WebSocket webSocket) {
+      synchronized (queuedMessages) {
+        client = webSocket;
+        connectionIsOpen = true;
+        queuedMessages.forEach(
+            message ->
+                client
+                    .sendText(message, true)
+                    .exceptionally(logWebSocketError(Level.SEVERE, "Failed to send queued text.")));
+        queuedMessages.clear();
+      }
+      webSocket.request(1);
+    }
+
+    @Override
+    public CompletionStage<?> onText(
+        final WebSocket webSocket, final CharSequence data, final boolean last) {
+      textAccumulator.append(data);
+      if (last) {
+        listener.messageReceived(textAccumulator.toString());
+        textAccumulator.setLength(0);
+      }
+      webSocket.request(1);
```

Comment:
This line perhaps could use a comment, why are we requesting to the websocket, and why `1`?